src/common/time/src/timestamp.rs

```diff
@@ -122,15 +180,8 @@ impl Timestamp {
     }
 
     pub fn to_chrono_datetime(&self) -> LocalResult<DateTime<Utc>> {
-        let nano_factor = TimeUnit::Second.factor() / TimeUnit::Nanosecond.factor();
-        let (mut secs, mut nsecs) = self.split();
-
-        if nsecs < 0 {
-            secs -= 1;
-            nsecs += nano_factor;
-        }
-
-        Utc.timestamp_opt(secs, nsecs as u32)
+        let (sec, nsec) = self.split();
+        Utc.timestamp_opt(sec, nsec)
```

Comment:
Why changing the logic?

Reply:
> Why changing the logic?

To remain consistent with chrono's [`NaiveDateTime`](https://docs.rs/chrono/0.4.23/chrono/naive/struct.NaiveDateTime.html#method.from_timestamp). We are going to use chrono to do the timestamp calculation, including add/sub etc.